### PR TITLE
Fix emit for shorthand properties when they refer to CommonJS exports.

### LINKED ISSUE #8826: Can't re-use exported variable in object literal shorthand (ES6/CommonJS)
**TypeScript Version:** 

Version 1.9.0-dev.20160525-1.0

**Code**

```ts
export const test = 'test'

export function foo () {
  run({ test })
}
```

```
tsc index.ts --target es6 --module commonjs
```

**Expected behavior:** Output using `exports.test` alias, or create a local `test` variable.

```ts
"use strict";
exports.test = 'test';
function foo() {
    run({ test: exports.test });
}
exports.foo = foo;
```

**Actual behavior:** Test is `undefined`.

```ts
"use strict";
exports.test = 'test';
function foo() {
    run({ test });
}
exports.foo = foo;
```

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Fixes #8826 so Yui can be more productive.

### CODE REVIEW

**Comment by yuit** [comment]
@DanielRosenwasser I will not merge it yet as this will have to be completely redo in transform branch

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -2150,9 +2150,9 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
             }
 
             // Return true if identifier resolves to an exported member of a namespace
-            function isNamespaceExportReference(node: Identifier) {
+            function isExportReference(node: Identifier) {
                 const container = resolver.getReferencedExportContainer(node);
-                return container && container.kind !== SyntaxKind.SourceFile;
+                return !!container;
```
Comment (mhegazy) [inline src/compiler/emitter.ts]: how can the container not be a sourceFile here?

Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: If you had a namespace, then the container is probably a namespace (i.e. `ModuleDeclaration` or `ModuleBlock`).

Are there any other containers I need to be aware of? Potentially statics on a class?

**Comment by mhegazy** [comment]
👍

**Comment by DanielRosenwasser** [comment]
@mhegazy feels that since this is an emit bug and that the fix is small enough, that it can be ported to the transforms branch relatively easily.

**Pull request merged**